### PR TITLE
feat(VBreadcrumbs): add item slot

### LINKED ISSUE #15742: [Feature Request] v-slot:item in v-breadcrumbs
### Problem to solve
So that I can add navigation by SPA mode
Example :
```
                   <v-breadcrumbs tag="ul" v-if="breadcrumbs.length" :items="breadcrumbs">
                        <template v-slot:prepend>
                            <v-icon size="small" icon="mdi-vuetify"></v-icon>
                        </template>
                        <template v-slot:item="{ item }">
                            <Link :href="item.href">

                            <v-breadcrumbs-item tag="li" :disabled="item.disabled">
                                {{ item.title.toUpperCase() }}
                            </v-breadcrumbs-item>
                            </Link>
                        </template>
                    </v-breadcrumbs>
```
As the Link component, through which I can navigate without refreshing the page (spa)

### Proposed solution
Add v-slot:item

### PR DESCRIPTION
**Pull request by samehdoush**
## Description
 fixes #15742

## Motivation and Context
Add v-slot:item to v-breadcrumbs

## Markup:

  ```
 <v-breadcrumbs tag="ul"  :items="breadcrumbs">
    <template v-slot:prepend>
        <v-icon size="small" icon="mdi-vuetify"></v-icon>
    </template>
    <template v-slot:item="{ item }">
      <v-breadcrumbs-item tag="li" :disabled="item.disabled">
          {{ item.title.toUpperCase() }}
      </v-breadcrumbs-item>
    </template>
</v-breadcrumbs>
  ```

### CODE REVIEW

**Comment by ilham89** [comment]
ok nice!

File: packages/vuetify/src/components/VBreadcrumbs/VBreadcrumbs.tsx [hunk 1]
```diff
@@ -110,15 +110,17 @@ export const VBreadcrumbs = genericComponent<new <T>() => {
...
-                key={ index }
-                disabled={ index >= array.length - 1 }
-                { ...(typeof item === 'string' ? { title: item } : item) }
-                v-slots={{
-                  default: slots.title ? () => slots.title?.({ item, index }) : undefined,
-                }}
-              />
-
+              { slots.item
+                ? slots.item?.({ item: item })
```
Comment (johnleider) [inline packages/vuetify/src/components/VBreadcrumbs/VBreadcrumbs.tsx:114]:
```suggestion
                ? slots.item.({ item, index })
```

File: packages/vuetify/src/components/VBreadcrumbs/VBreadcrumbs.tsx [hunk 2]
```diff
@@ -110,15 +110,17 @@ export const VBreadcrumbs = genericComponent<new <T>() => {
...
+                : (<VBreadcrumbsItem
+                  key={index}
+                  disabled={index >= array.length - 1}
+                  { ...(typeof item === 'string' ? { title: item } : item)}
+                  v-slots={{
+                    default: slots.title ? () => slots.title?.({ item, index }) : undefined,
+                  } }
+                />)
```
Comment (johnleider) [inline packages/vuetify/src/components/VBreadcrumbs/VBreadcrumbs.tsx:122]:
```suggestion
                : (
                  <VBreadcrumbsItem
                    key={index}
                    disabled={index >= array.length - 1}
                    { ...(typeof item === 'string' ? { title: item } : item)}
                    v-slots={{
                      default: slots.title ? () => slots.title.({ item, index }) : undefined,
                    } }
                  />
                )
```

**Review by johnleider — CHANGES_REQUESTED** [review]

File: packages/vuetify/src/components/VBreadcrumbs/VBreadcrumbs.tsx [hunk 3]
```diff
@@ -113,15 +113,19 @@ export const VBreadcrumbs = genericComponent<new <T>() => {
...
+                  <VBreadcrumbsItem
+                    key={index}
+                    disabled={index >= array.length - 1}
+                    { ...(typeof item === 'string' ? { title: item } : item)}
+                    v-slots={{
+                      default: slots.title ? () => slots.title.({ item, index }) : undefined,
+                    } }
+                  />
+                )
+              }
```
Comment (johnleider) [inline packages/vuetify/src/components/VBreadcrumbs/VBreadcrumbs.tsx:128]:
```suggestion
              { slots.item?.({ item, index }) ?? (
                  <VBreadcrumbsItem
                    key={ index }
                    disabled={ index >= array.length - 1 }
                    { ...(typeof item === 'string' ? { title: item } : item)}
                    v-slots={{
                      default: slots.title ? () => slots.title.({ item, index }) : undefined,
                    } }
                  />
                )
              }
```

**Review by johnleider — CHANGES_REQUESTED** [review]

**Review by johnleider — CHANGES_REQUESTED** [review]
My suggested commit wasn't spaced properly and I'm having issues cloning this for whatever reason. Could you resolve the lint issues please? Thank you

**Review by johnleider — APPROVED** [review]

**Pull request merged**